buzzmobile/tests/test_utils/readme.md

```diff
@@ -0,0 +1,258 @@
+# ROS Test Utils
+
+This library includes a number of tools for testing ros nodes in isolation
+without having to deal with the normal rostesting library.
+
+## Quick Notes
+
+ - If your tests are inexplicably failing, run `rm -r *.pyc`, especially if you
+   recently pulled or checked out a branch. The .pyc files are not managed by
+   vcs and will be used sometimes when they shouldn't be.
+
+## API
+
+The major components to `rostest_utils` are:
+
+ - `RosTest`
+ - `with_launch_file`
+ - `launch_node`
+ - `mock_pub`
+ - `check_topic`
+
+I'll go over each these in detail.
+
+#### `RosTest`
+
+This is a subclass of `unittest.Testcase`. It should be used for all
+integration-style tests that involve spinning up nodes and having them
+communicate via ROS. This class will automatically spin up an instance of Ros
+for each test you run, and make sure that it is isolated and cleaned up
+correctly.
+
+#### `with_launch_file`
+
+ROS requires launch files. Normally you would run something like `roslaunch
+xyz.launch`, but you can't do that when you aren't using the roslaunch command
+line tool, so this will do that for you. 
```

Comment:
can you include code examples for each of these? Like a single example per thing is fine, just to demonstrate